zebrad/src/commands/start.rs

```diff
@@ -54,6 +54,27 @@ impl StartCmd {
             config.network.network,
         ));
 
+        info!("starting legacy chain check");
+        match state
+            .clone()
+            .oneshot(zebra_state::Request::IsLegacyChain)
+            .await
+        {
+            Ok(zebra_state::Response::LegacyChain(response)) => {
+                if let Some(corrupt_db_path) = response {
+                    // we found a legacy chain
+                    panic!(
+                        "Legacy chain found, database can be corrupted.
+                        Delete your database and retry a full sync.
+                        Database path: {:?}",
+                        corrupt_db_path,
+                    );
```

Comment:
1. Usability: a state created by an older Zebra version isn't corrupt, it's just outdated. If we say "corrupt", users will think their disks are failing.

2. Format Bug: multi-line indented strings must be escaped in Rust: https://stackoverflow.com/questions/29483365/what-is-the-syntax-for-a-multiline-string-literal/29483453#29483453

3. Redundant comment: comments don't need to repeat what the code or error message says.

```suggestion
                if let Some(legacy_db_path) = response {
                    panic!(
                        "Cached state contains a legacy chain. \
                        An outdated Zebra version did not know about a recent network upgrade, \
                        so it followed a legacy chain using outdated rules. \
                        Hint: Delete your database, and restart Zebra to do a full sync. \
                        Database path: {:?}",
                        legacy_db_path,
                    );
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/2366/commits/d3989c6694db013997db3bbfcdc3d45e61f1142c